tasks/managed/push-snapshot/push-snapshot.yaml

```diff
@@ -307,50 +306,61 @@ spec:
             '.images[$i|tonumber] += {"arches": $arches, "oses": $oses, "name": $name, "shasum": $sha, "urls": []}' \
               "$RESULTS_JSON_FILE" > "$RESULTS_JSON_FILE.tmp" && mv "$RESULTS_JSON_FILE.tmp" "$RESULTS_JSON_FILE"
 
-          # Push source container if the component has pushSourceContainer: true or if the
-          # pushSourceContainer key is missing from the component and the defaults has
-          # pushSourceContainer: true
-          if [[ $(jq -r '.pushSourceContainer' <<< "$component") == "true" ]] \
-            || [[ $(jq 'has("pushSourceContainer")' <<< "$component") == "false" && \
-                  ${defaultPushSourceContainer} == "true" ]] ; then
-
-            source_tag=${origin_digest/:/-}.src
-            # Calculate the source container image based on the provided container image
-            sourceContainer="${source_repo}:${source_tag}"
-            # Check if the source container exists
-            source_container_digest=$(oras resolve --registry-config "$SOURCE_AUTH_FILE" \
-              "${sourceContainer}")
-
-            if [ -z "$source_container_digest" ] ; then
-              echo "Error: Source container ${sourceContainer} not found!"
-              exit 1
-            fi
-            # Push the source image with the source tag here. The source image will be
-            # pushed with the provided tags below in the loop
-            wait_for_slot
-            push_image "${source_container_digest}" "${name}" "${sourceContainer}" \
-              "${repository}" "${source_tag}" "" > "$TMP_RESULTS_DIR/${name}-${source_tag}.out" 2>&1 &
-            ((++REQUEST_COUNT))
-            echo "Request Count: $REQUEST_COUNT"
-          fi
-
-          for tag in $(jq -r '.[]' <<< "$imageTags") ; do
-            wait_for_slot
-            # Push the container image
-            push_image "${origin_digest}" "${name}" "${containerImage}" "${repository}" "${tag}" \
-            "$platform" > "$TMP_RESULTS_DIR/${name}-${tag}.out" 2>&1 &
-            ((++REQUEST_COUNT))
-            echo "Request Count: $REQUEST_COUNT"
-
-            # This variable will only exist if the above logic determined the source container should
-            # be pushed for this component
-            if [ -n "${source_container_digest-}" ] ; then
+          repositories=$(jq -c '.repositories' <<< "$component")
+          NUM_REPOS=$(jq 'length' <<< "$repositories")
+          
+          pushSourceContainer=$(jq -r '.pushSourceContainer' <<< "$component")
+          hasPushSourceContainer=$(jq 'has("pushSourceContainer")' <<< "$component")
+          
+          for ((j = 0; j < NUM_REPOS; j++)); do 
+            repository=$(jq -c --argjson j "$j" '.repositories[$j]' <<< "$component")
+            imageTags=$(jq '.tags // []' <<< "$repository")
```

Comment:
on line 237 you make sure each repository has tags. Here, you say use [] if no tags exist. I don't follow why we have to do that